src/test/java/org/reaktivity/reaktor/StreamsIT.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.reaktor;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.rules.RuleChain.outerRule;
+import static org.reaktivity.nukleus.route.RouteKind.SERVER;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+import org.reaktivity.nukleus.Configuration;
+import org.reaktivity.nukleus.Nukleus;
+import org.reaktivity.nukleus.NukleusBuilder;
+import org.reaktivity.nukleus.NukleusFactorySpi;
+import org.reaktivity.nukleus.stream.StreamFactoryBuilder;
+import org.reaktivity.reaktor.test.NukleusClassLoader;
+import org.reaktivity.reaktor.test.ReaktorRule;
+
+public class StreamsIT
+{
+    private final K3poRule k3po = new K3poRule()
+            .addScriptRoot("route", "org/reaktivity/specification/nukleus/control/route")
+            .addScriptRoot("streams", "org/reaktivity/specification/nukleus/streams");
+
+    private static StreamFactoryBuilder serverStreamFactory = new TestStreamFactoryBuilder();
```

Comment:
Should be using mock objects instead of "test" implementation code, right?

See [stack**overflow** question](https://stackoverflow.com/questions/28805688/in-java-how-can-i-mock-a-service-loaded-using-serviceloader) on how to mock a service using `ServiceLoader`.

Reply:
Are you suggesting to use PowerMockito and Mockito inside ReaktorRule and make those test dependencies that then all consumers of ReaktorRule would need to take on?

Reply:
I was initially thinking of doing it outside of `ReaktorRule`, but like your idea of moving it inside too.

Happy with either initially, please go with whatever is most expedient for now.

Reply:
Attempting to rewrite TestStreamFactory (especially TestStream) is very clumsy using mock objects, even using Mockito, and I'm not seeing the benefit. Can I leave TestStream as a concrete class? This is after all an integration test not a unit test.